### ISSUE TITLE
Installation always fails

### ISSUE THREAD

**Issue by balanit3** [elgatito/plugin.video.elementum#965]
Hi, I have installed build 21.0-BETA2 20.90.821 of kodi and tried to installl elemetum from repository but the install always fails. I have tried also 0.1.97 - linux-armv7-softfp and Gaia repository but fails too. 
I m using lg g2 with webos 7.3.1. I don’t know where look for log file

**Comment by elgatito** [comment]
@balanit3 
https://elementum.surge.sh/contribute/log/
or this https://kodi.wiki/view/Log_file/Easy

**Comment by jesterod** [comment]
it fails to open the repo 0.0.2 but after install from zip it fails to launch i get crash messages on the bottom
[kodi.log](https://github.com/elgatito/plugin.video.elementum/files/13807030/kodi.log)

**Comment by antonsoroko** [comment]
@jesterod from log:
>[WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions

i guess you use some [firewall](https://en.wikipedia.org/wiki/Firewall_(computing)) and it blocks elementum.
try to add kodi's and elementum's binaries to "allow list".

**Comment by jesterod** [comment]
Just windows defender on win 10 ive added exclusions for the elementum.exe in the bin/x64 and 86

**Comment by elgatito** [comment]
@jesterod You use some 3rd party repositories for installation. You should not do that.

```
CCurlFile::XFILE::CCurlFile::Open - <https://raw.githubusercontent.com/techecoyote/LooNaticsAsylum/master/matrix/zips/plugin.video.elementum/plugin.video.elementum-0.1.86.zip> Failed with code 404:
                                                   
2023-12-20 10:15:48.585 T:81064   error <general>: CAddonInstallJob[plugin.video.elementum]: failed to download special://home/addons/packages/plugin.video.elementum-0.1.86.zip
```

Download official repository and install from it: https://github.com/ElementumOrg/repository.elementumorg/releases/download/v0.0.2/repository.elementumorg-0.0.2.zip

**Comment by elgatito** [comment]
@jesterod Remove Elementum and that buggy repository, restart Kodi and re-install repository and Elementum.

**Comment by elgatito** [comment]
@jesterod This is the reason for your error:

```
2023-12-20 10:16:27.294 T:56776   error <general>: OSError
2023-12-20 10:16:27.294 T:56776   error <general>: : 
2023-12-20 10:16:27.294 T:56776   error <general>: [WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions
```

System does not allow Elementum to open port `65221`, but that is not a port opened by Elementum.exe, it is a port from Python part, which runs as part of Kodi process.

**Comment by jesterod** [comment]
> @jesterod This is the reason for your error:
> 
> ```
> 2023-12-20 10:16:27.294 T:56776   error <general>: OSError
> 2023-12-20 10:16:27.294 T:56776   error <general>: : 
> 2023-12-20 10:16:27.294 T:56776   error <general>: [WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions
> ```
> 
> System does not allow Elementum to open port `65221`, but that is not a port opened by Elementum.exe, it is a port from Python part, which runs as part of Kodi process.

So removing the other repo will fix this?

**Comment by elgatito** [comment]
@jesterod 

> So removing the other repo will fix this?

No. Your system is not allowing to open port `65221` to Elementum. That is why Elementum cannot start. 
It is not allowing to open port `65223` and that is why repository is not working.
There is nothing to fix from Kodi or Elementum. It is your system that does not allow.

**Comment by antonsoroko** [comment]
@jesterod the best bet is that you actually use firewall but you do not know about it (like it can be a part of some antivirus software, not windows defender - something else). or you have some security settings in windows that disallow elementum to open ports.

**Comment by jesterod** [comment]
Ive tried with firewall off and i dont run a antivirus…On Jan 10, 2024, at 12:40 PM, Anton Soroko ***@***.***> wrote:﻿
@jesterod the best bet is that you actually use firewall but you do not know about it (like it can be a part of some antivirus software, not windows defender - something else). or you have some security settings in windows that disallow elementum to open ports.

**Comment by antonsoroko** [comment]
@jesterod the only  2 things I can suggest are:
1. Try on different device
2. Reinstall Windows

**Comment by jesterod** [comment]
It works on my android device… reinstalling windows is such a pain to think of what to save and then remember what you forgot about when you go to use itOn Jan 11, 2024, at 4:44 AM, Anton Soroko ***@***.***> wrote:﻿
@jesterod the only  2 things I can suggest are:

Try on different device
Reinstall Windows

**Comment by antonsoroko** [comment]
@jesterod if you have "IT guy" friend - ask him to help to troubleshoot your Windows issue. Nothing we can do here. This is not elementum issue.

**Comment by antonsoroko** [comment]
@balanit3 if you are still interested - provide log files and also read 
https://github.com/elgatito/plugin.video.elementum/issues/939
https://github.com/elgatito/plugin.video.elementum/issues/951

those guys were able to run elementum on webos.

if not - we will close this issue.
